### DISCUSSION TITLE
Conditions working with multi-select picker

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#8966] [Help] [answered]
Hey,

I'm confused by the behaviour reported by one of my users, I tried multiple times and he's right :). Here is my setup:
I have a form that contains about 25 fields, some of those fields are displayed when required/needed.
e.g.:
There is a project field (multi-select picker), my user can select (one or many projects), then some of the fields are shown accordingly with this kind of rule:
![image](https://user-images.githubusercontent.com/91942877/206148827-de1f39a6-f35a-4ab0-a922-919162deb696.png)
I also tried with 'Is in' but without success.

If I don't select the right project, the field remains hidden.
If I select only the right project, the field is shown.
If I select the right project first and then other project,  the field is shown.
**If I select the other project and the right project, the field remains hidden ?**
If I then unselect the other project, the field is shown.

Did I miss something?
It seems that when the element is not at the beginning of the array, it does not work properly.

Thank you!

**Comment by oitkenmen** [comment]


**Reply by diogenesbrussels** [reply]
Hi, Glad to hear that you can do your application without this type of condition. What you're proposing here is a bit like "oh you have a flat tyre, have you tried rebuilding the car?"
The application is currently in use and I don't want to change the way it works for my users. I'm not going to change the usage of dozens of people to solve this kind of problem (unless no other solution is possible), that's not my vision of things.
I'm open to solutions that are constructive and trying to find why a simple condition is not behaving as expected.

**Comment by melohagan** [comment, marked as answer]
Hey @diogenesbrussels 

My guess is that the `Like` here is really just doing a *Starts With*.

I have an issue raised relating to the **Is in** filter: https://github.com/Budibase/budibase/issues/9015
In the meantime it should work if you use a JavaScript binding to explicitly return an array. Apologies for the late response!

It may be something like: 
```js
return [ "A", "B", "C" ];
```

**Reply by diogenesbrussels** [reply]
Hey @melohagan 

Thank you for the reply & explanation.
I solved this by using a js binding:
```
return Object.values($("action_form.Fields.action_type")).includes("Observation");
// equals True
```
because it is indeed a js object that is returned and not an array.

have a nice day!

**Reply by melohagan** [reply]
Nice one - thanks for sharing! Have a nice day 😄
